test/utils/IncomeSourcesUtilsSpec.scala

```diff
@@ -82,32 +85,30 @@ class IncomeSourcesUtilsSpec extends TestSupport with IncomeSourcesUtils {
   "removeIncomeSourceDetailsFromSession" when {
     "user has session data" should {
       "remove session data" in {
-        implicit val user = individualUser.copy()(fakeRequest)
+        implicit val user: MtdItUser[AnyContentAsEmpty.type] = individualUser.copy()(fakeRequest)
         val redirect = withIncomeSourcesRemovedFromSession {
           Redirect("nowhere")
-        }
+        }(individualUser, sessionService, ec)
 
-        redirect.session.get("addUkPropertyStartDate") shouldBe None
-        redirect.session.get("addBusinessName") shouldBe None
-        redirect.session.get("addBusinessTrade") shouldBe None
-        redirect.session.get("addIncomeSourcesAccountingMethod") shouldBe None
-        redirect.session.get("addBusinessStartDate") shouldBe None
-        redirect.session.get("addBusinessAccountingPeriodStartDate") shouldBe None
-        redirect.session.get("addBusinessAccountingPeriodEndDate") shouldBe None
-        redirect.session.get("addBusinessStartDate") shouldBe None
-        redirect.session.get("addBusinessAddressLine1") shouldBe None
-        redirect.session.get("addBusinessAddressLine2") shouldBe None
-        redirect.session.get("addBusinessAddressLine3") shouldBe None
-        redirect.session.get("addBusinessAddressLine4") shouldBe None
-        redirect.session.get("addBusinessPostalCode") shouldBe None
-        redirect.session.get("addBusinessCountryCode") shouldBe None
-        redirect.session.get("ceaseForeignPropertyDeclare") shouldBe None
-        redirect.session.get("ceaseForeignPropertyEndDate") shouldBe None
-        redirect.session.get("ceaseUKPropertyDeclare") shouldBe None
-        redirect.session.get("ceaseUKPropertyEndDate") shouldBe None
+        redirect.futureValue.session.get("addUkPropertyStartDate") shouldBe None
```

Comment:
sounds like a good candidate to iterate over list of fields defined in the SessionKeys again;
again, this is a massive code duplication